src/bin/pg_dump/dumpall_babel_utils.c

```diff
@@ -56,6 +81,82 @@ isBabelfishDatabase(PGconn *conn)
 	return (bbf_status == ON);
 }
 
+/*
+ * dumpBabelRestoreChecks:
+ * Dumps Babelfish specific pre-checks which get executed at the
+ * beginning of restore to validate if restore can be performed
+ * or not.
+ */
+void
+dumpBabelRestoreChecks(FILE *OPF, PGconn *conn, int binary_upgrade)
```

Comment:
can we avoid duplicate code by sharing utility function between dump and dumpall?

Reply:
Actually, pg_dump/pg_dumpall as standalone utilities and the function signature (arguments) and handling differs between both of them, so it might get over complicated to write a single function. Moreover, I think below piece of code is rarely going to change in future and eventually, we are going to remove it.